src/AutoMapper/TypeDetails.cs

```diff
@@ -125,40 +123,28 @@ where extensionMethod.IsGenericMethodDefinition
                     select extensionMethod.MakeGenericMethod(genericInterfaceArguments)
                 )
                 from methodMatch in matchedMethods
-                where methodMatch.GetParameters()[0].ParameterType.GetTypeInfo().IsAssignableFrom(genericInterface.GetTypeInfo())
+                where methodMatch.GetParameters()[0].ParameterType.IsAssignableFrom(genericInterface)
                 select methodMatch
             ).ToArray();
         }
 
-        private static MemberInfo[] BuildPublicReadAccessors(IEnumerable<MemberInfo> allMembers)
-        {
+        private static MemberInfo[] BuildPublicReadAccessors(IEnumerable<MemberInfo> allMembers) =>
             // Multiple types may define the same property (e.g. the class and multiple interfaces) - filter this to one of those properties
-            var filteredMembers = allMembers
+            allMembers
                 .OfType<PropertyInfo>()
                 .GroupBy(x => x.Name) // group properties of the same name together
                 .Select(x => x.First())
-                .Concat(allMembers.Where(x => x is FieldInfo)); // add FieldInfo objects back
-
-            return filteredMembers.ToArray();
-        }
+                .Concat(allMembers.Where(x => x is FieldInfo)) // add FieldInfo objects back
+                .ToArray();
 
-        private static MemberInfo[] BuildPublicAccessors(IEnumerable<MemberInfo> allMembers)
-        {
+        private static MemberInfo[] BuildPublicAccessors(IEnumerable<MemberInfo> allMembers) =>
             // Multiple types may define the same property (e.g. the class and multiple interfaces) - filter this to one of those properties
-            var filteredMembers = allMembers
+            allMembers
                 .OfType<PropertyInfo>()
                 .GroupBy(x => x.Name) // group properties of the same name together
-                .Select(x =>
-                    x.Any(y => y.CanWrite && y.CanRead)
-                        ? // favor the first property that can both read & write - otherwise pick the first one
-                        x.First(y => y.CanWrite && y.CanRead)
-                        : x.First())
-                .Where(pi => pi.CanWrite || pi.PropertyType.IsListOrDictionaryType())
```

Comment:
I changed this so we try to map now _all_ get-only collections. If you think is too impactful, I guess we can continue to say "backward compatibility" and keep the check.